### PR TITLE
CON-1125 contextual widget

### PR DESCRIPTION
**Pull request by shomavg**

### CODE REVIEW

File: src/Employer/Employer.Web/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -89,30 +89,31 @@
...
+            </a>
+        </div>
+        <div class="govuk-header__content">
+            <a href="@ExternalLinks.MaRoot" class="govuk-header__link govuk-header__link--service-name">
+                Manage apprenticeships
+            </a>
         </div>
-    </header>
+    </div>
+    @Html.Partial("_ZenDeskWidget")
```
Comment (Stephek) [inline src/Employer/Employer.Web/Views/Shared/_Layout.cshtml]: Don't put this in the header.

File: src/Employer/Employer.Web/Views/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -89,30 +89,31 @@
 
     @Html.Partial("_CookieBanner")
 
-    <header class="govuk-header " role="banner" data-module="header">
```
Comment (Stephek) [inline src/Employer/Employer.Web/Views/Shared/_Layout.cshtml]: Don't change the header.  It's being replaced with the global one.

File: src/Shared/Recruit.Shared.Web/Configuration/ZendeskConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Esfa.Recruit.Shared.Web.Configuration
+{
+    public sealed class ZendeskConfiguration
+    {
+        public string ZenDeskSnippetKey { get; set; }
```
Comment (Stephek) [inline src/Shared/Recruit.Shared.Web/Configuration/ZendeskConfiguration.cs]: Can these properties drop the ZenDesk prefix? They are in the ZendeskConfiguration class so it isn't needed.

File: src/Employer/Employer.Web/Startup.Configure.cs [hunk 4]
```diff
@@ -43,17 +44,50 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IOptions
...
                 app.UseHsts(hsts => hsts.MaxAge(365));
             }
-            
+
+            if (env.IsDevelopment())
+            {
+                IdentityModelEventSource.ShowPII = true;
+            }
             // Add Content Security Policy
+            app.Use(async (context, next) =>
```
Comment (Stephek) [inline src/Employer/Employer.Web/Startup.Configure.cs]: Can all the Zendesk config be put into it's own Use extension?  e.g. app.UseZenDesk ?

File: src/Employer/Employer.Web/Startup.Configure.cs [hunk 5]
```diff
@@ -43,17 +44,50 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IOptions
...
+            {
+                context.Response.Headers["Content-Security-Policy"] =
+                    "default-src 'self' 'unsafe-inline' https://*.zdassets.com https://*.zendesk.com wss://*.zendesk.com wss://*.zopim.com; " +
+                    "img-src 'self' *.google-analytics.com https://*.zdassets.com https://*.zendesk.com wss://*.zendesk.com wss://*.zopim.com; " +
+                    "script-src 'self' 'unsafe-inline' " +
+                    "*.googletagmanager.com *.postcodeanywhere.co.uk *.google-analytics.com *.googleapis.com https://*.zdassets.com https://*.zendesk.com wss://*.zendesk.com wss://*.zopim.com; " +
+                    "font-src 'self' data:;";
+                await next();
+            });
```
Comment (cofaulco) [inline src/Employer/Employer.Web/Startup.Configure.cs]: Do we need this middleware to add the header if we are already have the UseCsp middleware below setting all the same values?

File: azure/template.json [hunk 6]
```diff
@@ -915,6 +927,18 @@
...
                 "name": "PostcodeAnywhere:Key",
                 "value": "[parameters('PostcodeAnywhereKey')]"
+              },
+			  {
+                "name": "Zendesk:SnippetKey",
+                "value": "[parameters('ZendeskSnippetKey')]"
+              },
+			  {
+                "name": "Zendesk:SectionId",
+                "value": "[parameters('ZendeskSectionId')]"
```
Comment (cofaulco) [inline azure/template.json:937]: Might be best to name this ZendeskEmployerSectionId as they are different for employer and provider.

Comment (shomavg) [reply azure/template.json]: 👍

**Pull request merged**